src/lib/scan-imports.ts

```diff
@@ -0,0 +1,159 @@
+import fs from 'fs';
+import chalk from 'chalk';
+import ora from 'ora';
+import glob from 'glob';
+import {parse} from '@babel/parser';
+import traverse, {NodePath} from '@babel/traverse';
+import {
+  CallExpression,
+  ImportDeclaration,
+  isImportDeclaration,
+  isCallExpression,
+} from '@babel/types';
+
+/**
+ * Splits a filename into folder segments
+ */
+function explode(file: string) {
+  return file.split('/').filter(part => part);
+}
+
+/**
+ * Remove file extension
+ */
+function noExtension(file: string) {
+  return file.replace(/\.[^.]+$/, '');
+}
+
+/**
+ * Attempt to return npm package name
+ */
+function packageName(name: string) {
+  // if there’s no slash, this is likely a package name already
+  if (!name.includes('/')) {
+    return name;
+  }
+  // return first 2 segments if scoped package, or first if not scoped
+  const [scope, pkg] = explode(name);
+  return scope.startsWith('@') ? `${scope}/${pkg}` : scope;
+}
+
+/**
+ * Attempt to figure out web_module’s identity
+ */
+function resolveWebModule(name: string, dest: string) {
+  const lastSegment = [...explode(dest)].pop();
+  // is the last folder segment of --dest present?
+  if (name.includes(lastSegment)) {
```

Comment:
so, there's a bug here that actually may be a good solution to our prefix config problem :D 

Right now www.pika.dev hosts web modules out of `/static/web_modules` paths but this is routed out of a  `--dest web_modules` installed folder. So this technically shouldn't work, but because `'/static/web_modules'.includes('web_modules') === true`, detection still works. 

Which leads me to the idea: what if we just match URLs that include 'web_modules/', and strip out everything before and including web_modules. So we'd successfully match `/web_modules/*`, `static/web_modules/*`, and even `/path/to/my/custom/web_modules/*`. 

This is zero-config and straightforward, but the risk is that we match things in a pika-unrelated web_modules directory. But I don't think that's a real-life issue, since that has a meaning in @pika/web it would be strange to have a web_modules directory for something ELSE, and still using @pika/web.

What do you think?

Reply:
So this is what I’ve been racking my brain on for a while now! 🙂 I do think this is working as intended, though, because it only pays attention to the last segment of `--dest`. So if you had `--dest /path/to/pikachu`, it’d resolve `/pikachu/vue`.

It’s working as you described, but I think you’re proposing fixing it to `web_modules` which I don’t mind at all. I know a lot of `node_modules` resolution that happens in a number of different things are complicated, but they all kinda hinge on `node_modules` being part of the name. I think we could do the same, because I agree with you—I’ve never seen anything with `web_modules` as part of its normal path, and I can’t think of a scenario where you’d be using `web_modules` in a way that Pika shouldn’t know about.

So I guess my question is: should we keep the behavior of the last segment of `--dest` (current), or simplify it and go with `web_modules`? I’m OK with the latter; just wanted to confirm.

Reply:
OK you’ve sold me. And thanks for talking this through.

I think it’s helpful to trace that tradeoff of flexibility (works for weird setups) with more config vs opinionated and less config. I know we’re spitballing at usage now but I agree that catering a scenario where someone **isn’t** using `web_modules` as part of their path is something that’s rare enough now that it’s probably not worth planning for. Either way I’m just happy we didn’t make an individual CLI flag for this.